src/Connection/Dsn.php

```diff
@@ -66,4 +72,26 @@ public function getDriver(): string
     {
         return $this->driver;
     }
+
+    private function buildDsn(): string
+    {
+        $dsn = "$this->driver:" . "host=$this->host" . ';' . "dbname=$this->databaseName";
+
+        if ($this->port !== null) {
+            $dsn .= ';' . "port=$this->port";
+        }
+
+        return $dsn;
+    }
+
+    private function buildDsnSqlSrv(): string
```

Comment:
If there's a difference like that, it should be on each driver's side about forming a string from a set of options.